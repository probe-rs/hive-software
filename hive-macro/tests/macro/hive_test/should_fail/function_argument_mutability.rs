#[hive_macro::hive_test]
fn test_function(test_channel: &String, handle: String, info: String) {}

fn main() {}
