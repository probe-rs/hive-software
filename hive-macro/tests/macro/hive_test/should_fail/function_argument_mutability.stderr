error: Function argument should be mutable

         = help: This argument in a Hive testfunction is a mutable reference.
         = note: example: fn my_fancy_test(test_channel: &mut dyn TestChannelHandle, session: &mut Session, target_info: &HiveTargetInfo);

 --> tests/macro/hive_test/should_fail/function_argument_mutability.rs:2:32
  |
2 | fn test_function(test_channel: &String, handle: String, info: String) {}
  |                                ^
